rskj-core/src/main/java/org/ethereum/config/SystemProperties.java

```diff
@@ -705,4 +705,50 @@ private Optional<List<BtcECKey>> getGenesisFederationPublicKeys() {
     public long getGasEstimationCap() {
         return configFromFiles.getLong(PROPERTY_RPC_GAS_ESTIMATION_CAP);
     }
+
+    public DbKind databaseKind() {
+        return DbKind.ofName(configFromFiles.getString("keyvalue.datasource"));
+    }
+
+    public DbKind getDbKindValueFromDbKindLogFile() {
+        try {
+            File file = new File(databaseDir(), "dbKind.properties");
+            Properties props = new Properties();
+            if (file.exists() && file.canRead()) {
+                try (FileReader reader = new FileReader(file)) {
+                    props.load(reader);
+                }
+
+                return DbKind.ofName(props.getProperty("keyvalue.datasource"));
+            }
+
+            return null;
+        } catch (IOException e) {
+            return null;
```

Comment:
do you think we could propagate this exception up the stack?

Reply:
I don't think it is needed at all, since I'm just expecting to get the value from the fiile, if the file doesn't exist I would just return null, because the user can remove it at any moment.